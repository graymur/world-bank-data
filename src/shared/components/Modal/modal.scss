@import "~style/lib";

.wb-modal {
    position: fixed;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    z-index: 999;
    background: rgba(255, 255, 255, 0.75);

    &__content {
        position: absolute;
        border: 1px solid rgb(204, 204, 204);
        background: rgb(255, 255, 255);
        overflow: auto;
        border-radius: 4px;
        outline: none;
        padding: 2rem;

        @include mobile() {
            top: 2rem;
            left: 2rem;
            right: 2rem;
            bottom: 2rem;
        }

        @include min-screen($tablet) {
            top: 4rem;
            left: 4rem;
            right: 4rem;
            bottom: 4rem;
        }
    }
}