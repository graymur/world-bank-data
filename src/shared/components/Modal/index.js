import React from 'react';
import PropTypes from 'prop-types';
import './modal.scss';

const Modal = ({children}) => {
	return (
		<div className='wb-modal'>
			<div className='wb-modal__content'>
				{children}
			</div>
		</div>
	);
};

Modal.propTypes = {
	children: PropTypes.node
};

export default Modal;
