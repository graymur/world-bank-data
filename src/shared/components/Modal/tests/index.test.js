/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import Modal from '../index';

test('Renders children in wrapper', () => {
	const component = renderer.create(<Modal><h1>content</h1></Modal>);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
