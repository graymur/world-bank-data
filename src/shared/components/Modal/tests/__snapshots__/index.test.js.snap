// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders children in wrapper 1`] = `
<div
  className="wb-modal"
>
  <div
    className="wb-modal__content"
  >
    <h1>
      content
    </h1>
  </div>
</div>
`;
